test/extensions/tracers/zipkin/zipkin_tracer_impl_test.cc

```diff
@@ -1068,6 +1068,389 @@ TEST_F(ZipkinDriverTest, DuplicatedHeader) {
+      - header:
+          key: "Authorization"
+          value: "Bearer token123"
+      - header:
+          key: "X-Custom-Header"
+          value: "custom-value"
+      - header:
+          key: "X-API-Key"
+          value: "api-key-123"
+  collector_endpoint_version: HTTP_JSON
+  )EOF";
+
+  envoy::config::trace::v3::ZipkinConfig zipkin_config;
+  TestUtility::loadFromYaml(yaml_string, zipkin_config);
+  setup(zipkin_config, true);
+
+  Http::MockAsyncClientRequest request(&cm_.thread_local_cluster_.async_client_);
+  Http::AsyncClient::Callbacks* callback;
+  const absl::optional<std::chrono::milliseconds> timeout(std::chrono::seconds(5));
+
+  // Set up expectations for the HTTP request with custom headers
+  EXPECT_CALL(cm_.thread_local_cluster_.async_client_,
+              send_(_, _, Http::AsyncClient::RequestOptions().setTimeout(timeout)))
+      .WillOnce(
+          Invoke([&](Http::RequestMessagePtr& message, Http::AsyncClient::Callbacks& callbacks,
+                     const Http::AsyncClient::RequestOptions&) -> Http::AsyncClient::Request* {
+            callback = &callbacks;
+
+            // Verify standard headers are present
+            EXPECT_EQ("/api/v2/spans", message->headers().getPathValue());
+            EXPECT_EQ("zipkin-collector.example.com", message->headers().getHostValue());
+            EXPECT_EQ("application/json", message->headers().getContentTypeValue());
+
+            // Verify custom headers are present
+            auto auth_header = message->headers().get(Http::LowerCaseString("authorization"));
+            EXPECT_FALSE(auth_header.empty());
+            EXPECT_EQ("Bearer token123", auth_header[0]->value().getStringView());
+
+            auto custom_header = message->headers().get(Http::LowerCaseString("x-custom-header"));
+            EXPECT_FALSE(custom_header.empty());
+            EXPECT_EQ("custom-value", custom_header[0]->value().getStringView());
+
+            auto api_key_header = message->headers().get(Http::LowerCaseString("x-api-key"));
+            EXPECT_FALSE(api_key_header.empty());
+            EXPECT_EQ("api-key-123", api_key_header[0]->value().getStringView());
+
+            return &request;
+          }));
+
+  EXPECT_CALL(runtime_.snapshot_, getInteger("tracing.zipkin.min_flush_spans", 5))
+      .WillOnce(Return(1));
+  EXPECT_CALL(runtime_.snapshot_, getInteger("tracing.zipkin.request_timeout", 5000U))
+      .WillOnce(Return(5000U));
+
+  Tracing::SpanPtr span = driver_->startSpan(config_, request_headers_, stream_info_,
+                                             operation_name_, {Tracing::Reason::Sampling, true});
+  span->finishSpan();
+
+  Http::ResponseHeaderMapPtr response_headers{
+      new Http::TestResponseHeaderMapImpl{{":status", "202"}}};
+  callback->onSuccess(request,
+                      std::make_unique<Http::ResponseMessageImpl>(std::move(response_headers)));
+}
+
+TEST_F(ZipkinDriverTest, ReporterFlushWithHttpServiceHeadersProtobuf) {
```

Comment:
This test is unnecessary because the `collector_service` has no business with the body's type (json or protobuf).